scripts/upload-and-link-plugin-zip.mjs

```diff
@@ -0,0 +1,97 @@
+import chalk from 'chalk';
+import { execSync } from "node:child_process";
+import process from "node:process";
+import { parseArgs } from "node:util";
+
+const { env } = process;
+
+const { values: { pr, commit, merged } } = parseArgs( {
+	options: {
+		pr: { type: "string" },
+		commit: { type: "string" },
+		merged: { type: "boolean", default: false },
+	},
+} );
+
+const MEDIA_LIBRARY_ENDPOINT = "https://wpjobmanager.com/wp-json/wp/v2/media"
+
+const login = env.WPJMCOM_API_LOGIN;
+
+( function main() {
+
+	// deleteOldZips();
+	//
+	// if ( merged ) {
+	// 	return;
+	// }
+	//
+	// const zip = uploadZip();
+	const zip = "https://wpjobmanager.com/wp-content/uploads/2023/09/wp-job-manager-zip-2577-26255dc9.zip";
+	addComment( zip );
+
+} )();
+
+/**
+ * Remove previously uploaded plugin zip files for the PR.
+ *
+ */
+function deleteOldZips() {
+
+	const oldZips = JSON.parse( execSync( `curl -s -u "${ login }" "${ MEDIA_LIBRARY_ENDPOINT }?mime_type=application/zip&_fields=id,date,title,source_url"` ).toString() );
+	if ( oldZips?.code || ! Array.isArray( oldZips ) ) {
+		console.log( `[${ oldZips.code }]`, oldZips.message );
+		return;
+	}
+	oldZips?.forEach( ( zip ) => {
+		const
+			title = zip.title.rendered;
+		if ( title.match( new RegExp( `wp-job-manager-zip-${pr}` ) ) ) {
+			console.log( `Deleting old plugin build ${ title }.zip` );
+			execSync( `curl -s -u "${ login }" -X DELETE "${ MEDIA_LIBRARY_ENDPOINT }/${ zip.id }?force=true"` );
+		}
+	} )
+}
+
+/**
+ * Upload plugin zip to media library.
+ *
+ * @returns {string} URL to plugin zip file.
+ */
+function uploadZip() {
+
+	const id = `${ pr }-${ commit.substring( 0, 8 ) }`;
+
+	const zip = execSync( `curl -u "${ login }" --http1.1 --data-binary @wp-job-manager.zip -H "Content-Disposition: attachment; filename=\"wp-job-manager-zip-${ id }.zip\"" ${ MEDIA_LIBRARY_ENDPOINT }?title=wp-job-manager-zip-${ id } | jq .source_url` ).toString().replaceAll( '"', '' ).trim();
```

Comment:
If a non-automattician forks this repo, and then send a PR to it, then `login` will be empty, as [the secret isn't available in these scenarios](https://docs.github.com/en/actions/managing-workflow-runs/approving-workflow-runs-from-public-forks). Maybe we shold avoid trying to run the `curl` command if `login` is empty? (and then avoid modifying the description too, naturally?)

Also, what happens if this command fails to run properly?

Reply:
Good catch, I added an early exit. Generally if anything throws an error it should make it to the workflow's logs, but made it a bit more graceful ([`commits`](https://github.com/Automattic/WP-Job-Manager/pull/2577/files/64e63fba8aa437c5e6be6606af534ea97df47282..265b709cac8676e43712f1818d30f552b8bd8876))